### PR TITLE
use clog, our fork of glog

### PR DESCRIPTION
**Pull request by tbg**
Currently this should behave identically to glog but we can now use our own version of `V(.)`.
The plan is:
- [x] add switch to `glog` that omits the header (so it prints only your log message)
- [ ] add functions `log.Infoc(ctx context.Context, msg)` which will wrap around `clog.Print(f?)`
- [ ] clean up the other formatted loggers. We shouldn't be logging much unstructured data at all, so it may be a good idea to remove all log functions `log\.\w+f`. 
- [ ] switch away from flag handling, or at least switch to pflags.

Partakers of this PR should also check out the relevant commit https://github.com/cockroachdb/clog/commit/d2542376307a54ad0ff768be384c3c589ffacca1 on `clog`. From now on I'll do PRs there as well.

### CODE REVIEW

**Comment by tschottdorf** [comment]
I'll move the TODO list into issues once I'm done here, btw.

File: util/sampling.go [hunk 1]
```diff
@@ -99,7 +99,7 @@ func (rs *WeightedReservoirSample) Consider(value interface{}) {
 // computing x**1/weight may be ill-behaved.
 func (rs *WeightedReservoirSample) ConsiderWeighted(value interface{}, weight float64) {
 	if weight <= 0 {
-		glog.Warningf("reservoir sample received non-positive weight %f", weight)
+		clog.Warningf("reservoir sample received non-positive weight %f", weight)
```
Comment (petermattis) [inline util/sampling.go]: Should this be using `cockroachdb/cockroach/util/log`?

**Comment by petermattis** [comment]
LGTM

File: util/sampling.go (hunk 1 above)
Comment (tschottdorf) [inline util/sampling.go]: yes. I'll fix it. And by "fix" I mean I will delete the file since we're not using it anyways.

**Pull request merged**